misk-action-scopes/src/main/kotlin/misk/scope/ActionScope.kt

```diff
@@ -57,12 +57,14 @@ class ActionScope @Inject internal constructor(
   }
 
   /** Starts the scope on a thread with the provided seed data */
-  fun enter(seedData: Map<Key<*>, Any?>): ActionScope {
+  override fun enter(seedData: Map<out Any, Any?>): Scope {
```

Comment:
I do worry about the compatibility layer approach if the end result is erasing lots of types with `Any`. That seems to result in net worse APIs and usage for the slim benefit of not including an extra library in Armeria projects. 

How would a future Misk user know that this `enter(seedData)` parameter isn't actually `Map<Any, Any>` but immediately is cast to `Key` until their code surprisingly blows up in tests or runtime on cast failure?